pkg/kv/mem/store.go

```diff
@@ -111,6 +119,9 @@ func (s *Store) SetIf(_ context.Context, partitionKey, key, value []byte, valueP
 	}
 	s.mu.Lock()
 	defer s.mu.Unlock()
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
```

Comment:
```suggestion
	if ctx.Err() != nil {
		return ctx.Err()
	}
	s.mu.Lock()
	defer s.mu.Unlock()
```

Don't think we should (possibly) wait for the lock just to find out ctx is cancelled.
It will also keep it consistent with the rest of the functions in this file and prevent future quesiotns

Reply:
Done.